packages/jest-matchers/src/index.js

```diff
@@ -10,27 +10,37 @@
 
 'use strict';
 
-import type {ThrowingMatcherFn, ExpectationResult} from '../types';
+import type {
+  Expect,
+  ExpectationResult,
+  ExpectationObject,
+  MatchersObject,
+  RawMatcherFn,
+  ThrowingMatcherFn,
+} from '../types';
 
 const matchers = require('./matchers');
+const GLOBAL_MATCHERS_OBJECT_KEY = '_JEST_MATCHERS';
+global[GLOBAL_MATCHERS_OBJECT_KEY] = {};
```

Comment:
Can you explain how you are hoping to use addMatchers in Jest? I'm quite a bit against adding new globals to Jest.

I'd prefer to re-use `jest.addMatchers` by making it compatible with both Jasmine2 and the new matcher system. We don't use `jest.addMatchers` at FB, so it is a good chance to start using it. It's ok if we are breaking the API a bit, all your matchers work requires us to go to 14.0 soon anyway and this can easily be codemodded.

Regarding the global, I am also hesitant on storing it on global. Number 3 is a strict downside. I don't want anyone to modify or read this and opening it up to this will just make someone frown in five years when they have to clean up a gigantic mess at FB because some people started abusing this. I think your #2 point is valid, however I first need to see how addMatchers is being used. If it's added from the outside context in (through `jest.addMatchers`), the module registry will not be an issue.

Finally, if we do decide this must absolutely be global for some reason, then I would like us to use

```js
Object.defineProperty(global, Symbol.for('$$jest-add-matchers'), {
  // these are all defaults:
  // configurable: false,
  // enumerable: false,
  // writable: false,
  value: {…}
})
```

so that it is unlikely people will mess with it.